### PR TITLE
Refactor react-router

### PR DESCRIPTION
**Pull request by pingustar**

### CODE REVIEW

File: src/components/core/menu/mainMenu/MainMenuLeft.tsx [hunk 1]
```diff
@@ -16,23 +16,35 @@ export const MainMenuLeft: FC = () => {
...
+        {menuItems.map(({ label, href }, index) => {
+          let isSamePage = match({
+            to: href,
+            search: {},
+            params: {},
+            fuzzy: true,
+          });
+          if (pathname.startsWith('/strategies') && href === '/') {
+            isSamePage = true;
+          }
```
Comment (GrandSchtroumpf) [inline src/components/core/menu/mainMenu/MainMenuLeft.tsx:35]: Might be more readable to extract think into a dedicated function outside of the template.
```typescript
const isSamePageLink = (to: string) => {
  if (pathname.startsWith('/strategies') && href === '/') return true;
  return match({ to, search: {}, params: {}, fuzzy: true });
} 
...
{menuItems.map(({ label, href }, index) => {
  const isSamePage = isSamePageLink(href);
})
```

File: src/components/explorer/ExplorerSearchDropdownItems.tsx [hunk 2]
```diff
@@ -28,7 +28,8 @@ export const ExplorerSearchDropdownItems: FC<Props> = ({ setSearch }) => {
       {items.map(({ label, type, active }) => (
         <Link
           key={type}
-          to={PathNames.explorer(type)}
+          to={'/explorer/$type'}
```
Comment (GrandSchtroumpf) [inline src/components/explorer/ExplorerSearchDropdownItems.tsx:31]:
```suggestion
          to="/explorer/$type"
```

File: src/components/explorer/useExplorerParams.ts [hunk 3]
```diff
@@ -1,8 +1,11 @@
 import { useParams } from 'libs/routing';
-import { ExplorerParams } from './utils';
 
 export const useExplorerParams = () => {
-  const params: ExplorerParams = useParams({ strict: false });
+  // TODO broken type inference
+  // const params = useParams({ from: explorerResultLayout.fullPath });
```
Comment (GrandSchtroumpf) [inline src/components/explorer/useExplorerParams.ts:5]: Can we remove the comments ?

File: src/components/strategies/StrategyPageTabs.tsx [hunk 4]
```diff
@@ -15,16 +15,27 @@ interface Props {
...
+          fuzzy:
+            currentPathname.includes('/token/') && href.includes('portfolio'),
+        });
+
         return (
           <Link
             to={href}
+            // TODO: fix this
+            params={params ?? {}}
+            search={{}}
```
Comment (GrandSchtroumpf) [inline src/components/strategies/StrategyPageTabs.tsx:38]: Same as above. Do we need an issue for that ?

File: src/components/strategies/create/CreateStrategyCTA.tsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ import { buttonStyles } from 'components/common/button/buttonStyles';
 export const CreateStrategyCTA = () => {
   return (
     <Link
-      to={PathNames.createStrategy}
+      to={'/strategies/create'}
```
Comment (GrandSchtroumpf) [inline src/components/strategies/create/CreateStrategyCTA.tsx:11]:
```suggestion
      to="/strategies/create"
```

File: src/components/strategies/create/CreateStrategyCTA.tsx [hunk 6]
```diff
@@ -33,7 +33,7 @@ export const CreateStrategyCTAMobile = () => {
 
   return (
     <Link
-      to={PathNames.createStrategy}
+      to={'/strategies/create'}
```
Comment (GrandSchtroumpf) [inline src/components/strategies/create/CreateStrategyCTA.tsx:36]:
```suggestion
      to="/strategies/create"
```

File: src/components/strategies/overview/strategyBlock/StrategyBlockCreate.tsx [hunk 7]
```diff
@@ -15,7 +15,7 @@ export const StrategyBlockCreate: FC<Props> = ({
   return (
     <Link
       onClick={() => carbonEvents.strategy.newStrategyCreateClick(undefined)}
-      to={PathNames.createStrategy}
+      to={'/strategies/create'}
```
Comment (GrandSchtroumpf) [inline src/components/strategies/overview/strategyBlock/StrategyBlockCreate.tsx:18]:
```suggestion
      to="/strategies/create"
```

File: src/components/strategies/portfolio/PortfolioMobileCard.tsx [hunk 8]
```diff
@@ -40,7 +42,12 @@ export const PortfolioMobileCard: FC<Props> = ({
 
   if (href) {
     return (
-      <Link to={href} className={wrapperClasses}>
+      <Link
```
Comment (GrandSchtroumpf) [inline src/components/strategies/portfolio/PortfolioMobileCard.tsx:45]: When should we use `Link` over `A` ?

File: src/components/core/error/ErrorUserBlocked.tsx [hunk 9]
```diff
@@ -17,14 +17,9 @@ export const ErrorUserBlocked = () => {
       variant={'error'}
     >
       <div className={'space-y-10'}>
-        <Link
-          to={
-            'https://home.treasury.gov/policy-issues/financial-sanctions/recent-actions/20220808'
-          }
-          className={'w-full'}
-        >
+        <A to={externalLinks.treasuryGov} className={'w-full'}>
```
[the remark this answers is not in the document]
Comment (pingustar) [reply src/components/core/error/ErrorUserBlocked.tsx:20]: I know, my IDE does this automatically.

File: src/libs/modals/modals/ModalConfirmStrategy/ModalConfirmDelete.tsx [hunk 10]
```diff
@@ -66,7 +66,7 @@ export const ModalConfirmDelete: ModalFC<ModalConfirmDeleteData> = ({
           <Link
             onClick={editPrices}
             disabled={isAwaiting || isProcessing}
-            to={PathNames.editStrategy}
+            to={'/strategies/edit/$strategyId'}
```
Comment (GrandSchtroumpf) [inline src/libs/modals/modals/ModalConfirmStrategy/ModalConfirmDelete.tsx:69]:
```suggestion
            to="/strategies/edit/$strategyId"
```

File: src/libs/modals/modals/ModalConfirmStrategy/ModalConfirmWithdraw.tsx [hunk 11]
```diff
@@ -40,7 +40,7 @@ export const ModalConfirmWithdraw: ModalFC<ModalConfirmWithdrawData> = ({
         <h3 className="text-14 font-weight-500">Did you know ?</h3>
         <Link
           onClick={edit}
-          to={PathNames.editStrategy}
+          to={'/strategies/edit/$strategyId'}
```
Comment (GrandSchtroumpf) [inline src/libs/modals/modals/ModalConfirmStrategy/ModalConfirmWithdraw.tsx:43]:
```suggestion
          to="/strategies/edit/$strategyId"
```

File: src/libs/modals/modals/ModalConfirmStrategy/ModalConfirmWithdraw.tsx [hunk 12]
```diff
@@ -56,7 +56,7 @@ export const ModalConfirmWithdraw: ModalFC<ModalConfirmWithdrawData> = ({
       </article>
       <Link
         onClick={edit}
-        to={PathNames.editStrategy}
+        to={'/strategies/edit/$strategyId'}
```
Comment (GrandSchtroumpf) [inline src/libs/modals/modals/ModalConfirmStrategy/ModalConfirmWithdraw.tsx:59]:
```suggestion
        to="/strategies/edit/$strategyId"
```

File: src/libs/modals/modals/WalletModal/ModalWalletContent.tsx [hunk 13]
```diff
@@ -40,7 +40,7 @@ export const ModalWalletContent: FC<Props> = ({ onClick, isLoading }) => {
         <p>
           By connecting my wallet, I agree to the{' '}
           <Link
-            to={PathNames.terms}
+            to={'/terms'}
```
Comment (GrandSchtroumpf) [inline src/libs/modals/modals/WalletModal/ModalWalletContent.tsx:43]:
```suggestion
            to="/terms"
```

File: src/libs/modals/modals/WalletModal/ModalWalletContent.tsx [hunk 14]
```diff
@@ -49,7 +49,7 @@ export const ModalWalletContent: FC<Props> = ({ onClick, isLoading }) => {
           and{' '}
           <Link
             target="_blank"
-            to={PathNames.privacy}
+            to={'/privacy'}
```
Comment (GrandSchtroumpf) [inline src/libs/modals/modals/WalletModal/ModalWalletContent.tsx:52]:
```suggestion
            to="/privacy"
```

File: src/libs/routing/A.tsx [hunk 15]
```diff
@@ -0,0 +1,11 @@
+import { FC, HTMLProps } from 'react';
+
+type Props = HTMLProps<HTMLAnchorElement> & { to: string };
+
+export const A: FC<Props> = ({ to, children, ...props }) => {
```
Comment (GrandSchtroumpf) [inline src/libs/routing/A.tsx:5]: I think the name doesn't reflect the logic. Maybe something like `NewTabLink` or something like.

File: src/libs/routing/routes/explorer.ts [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+    });
+  },
+});
+
+export const explorerPage = new Route({
+  getParentRoute: () => explorerLayout,
+  path: '$type',
+  parseParams: (params: Record<string, string>) => {
+    return { type: params.type as ExplorerType };
+  },
```
Comment (GrandSchtroumpf) [inline src/libs/routing/routes/explorer.ts:32]: couldn't be rewritten like that ? 
```suggestion
  parseParams: <T extends { type:  ExplorerType}>({ type }: T) => { type },
```

File: src/components/strategies/portfolio/PortfolioMobileCard.tsx (hunk 8 above)
Comment (pingustar) [reply src/components/strategies/portfolio/PortfolioMobileCard.tsx:45]: A is just a wrapper around the native anchor element with some defaults. Use this for any external links.
Link is the react-router component for internal navigation.

I didnt get type-safety to work properly with the combo component we had.

File: src/libs/routing/routes/explorer.ts (hunk 16 above)
Comment (pingustar) [reply src/libs/routing/routes/explorer.ts:32]: this doesn't seem to work unfortunately. 
telegram-cloud-photo-size-1-4994821896533028042-y

File: src/components/strategies/StrategyPageTabs.tsx (hunk 4 above)
Comment (pingustar) [reply src/components/strategies/StrategyPageTabs.tsx:38]: #998

File: src/components/core/menu/mainMenu/MainMenuLeft.tsx [hunk 17]
```diff
@@ -16,23 +16,35 @@ export const MainMenuLeft: FC = () => {
...
+
           return (
             <Link
               key={index}
               onClick={() => handleOnItemClick(href)}
               to={href}
+              // TODO: fix this
```
[the remark this answers is not in the document]
Comment (pingustar) [reply src/components/core/menu/mainMenu/MainMenuLeft.tsx:39]: #998

File: src/components/core/menu/mainMenu/MainMenuLeft.tsx [hunk 18]
```diff
@@ -16,20 +21,24 @@ export const MainMenuLeft: FC = () => {
...
         onClick={() => carbonEvents.navigation.navHomeClick(undefined)}
       >
         <LogoCarbon className={'w-34'} />
       </Link>
 
       <div className={'hidden space-x-24 md:block'}>
-        {menuItems.map(({ label, href, hrefMatches }, index) => {
-          const isSamePage = isPathnameMatch(pathname, href, hrefMatches);
+        {menuItems.map(({ label, href }, index) => {
+          let isSamePage = isSamePageLink(href);
```
Comment (GrandSchtroumpf) [inline src/components/core/menu/mainMenu/MainMenuLeft.tsx:32]: You can use `const` now
```suggestion
          const isSamePage = isSamePageLink(href);
```

File: src/components/core/menu/mobileMenu/MobileMenu.tsx [hunk 19]
```diff
@@ -1,30 +1,41 @@
...
-            isPathnameMatch(location.pathname, href, hrefMatches)
-              ? 'text-white'
-              : 'hover:text-white'
-          }`}
-        >
-          {label}
-        </Link>
-      ))}
+      {menuItems.map(({ label, href }, index) => {
+        let isSamePage = isSamePageLink(href);
```
Comment (GrandSchtroumpf) [inline src/components/core/menu/mobileMenu/MobileMenu.tsx:20]:
```suggestion
        const isSamePage = isSamePageLink(href);
```

File: src/components/strategies/portfolio/PortfolioMobileCard.tsx [hunk 20]
```diff
@@ -40,7 +42,12 @@ export const PortfolioMobileCard: FC<Props> = ({
...
+        to={href}
+        params={params ?? {}}
```
Comment (GrandSchtroumpf) [inline src/components/strategies/portfolio/PortfolioMobileCard.tsx:47]: Should we add a `TODO: fix it` here too ?

**Review by GrandSchtroumpf — APPROVED** [review]
Very nice work 👍

File: src/components/strategies/portfolio/PortfolioMobileCard.tsx (hunk 20 above)
Comment (pingustar) [reply src/components/strategies/portfolio/PortfolioMobileCard.tsx:48]: jupp

File: src/components/core/MainContent.tsx [hunk 21]
```diff
@@ -10,11 +10,17 @@ import { carbonEvents } from 'services/events';
...
 import {
   Outlet,
-  PathNames,
+  Pathnames,
   useRouterState,
   useScrollRestoration,
 } from 'libs/routing';
 
+const paths: Record<string, Pathnames> = {
+  debug: '/debug',
```
Comment (zavelevsky) [inline src/components/core/MainContent.tsx:19]: why do we have path strings everywhere? Having type safety doesn't mean it's a good idea to write hard coded strings in more than one place

File: src/components/core/MainContent.tsx [hunk 22]
```diff
@@ -32,9 +38,9 @@ export const MainContent: FC = () => {
   }, [location, location.pathname]);
 
   if (
-    location.pathname === PathNames.debug ||
-    location.pathname === PathNames.terms ||
-    location.pathname === PathNames.privacy
+    location.pathname === paths.debug ||
```
Comment (zavelevsky) [inline src/components/core/MainContent.tsx:41]: Object.values(paths).includes(location.pathname)

File: src/components/core/menu/mainMenu/MainMenuLeft.tsx [hunk 23]
```diff
@@ -1,13 +1,18 @@
...
-import { isPathnameMatch } from 'utils/helpers';
 import { handleOnItemClick } from '../utils';
 import { menuItems } from 'components/core/menu';
 
 export const MainMenuLeft: FC = () => {
-  const pathname = useRouterState().location.pathname;
+  const { pathname } = useRouterState().location;
+  const match = useMatchRoute();
+
+  const isSamePageLink = (to: string) => {
```
Comment (zavelevsky) [inline src/components/core/menu/mainMenu/MainMenuLeft.tsx:12]: this looks like some utility for libs/routing

File: src/components/core/menu/mainMenu/MainMenuRightBurger/useBurgerMenuItems.tsx [hunk 24]
```diff
@@ -38,66 +37,66 @@ export const useBurgerMenuItems = () => {
...
+        <NewTabLink className="flex" to={externalLinks.blog}>
           Blog
-        </Link>
+        </NewTabLink>
       ),
     },
     {
       content: (
-        <Link className="flex" to={PathNames.terms}>
+        <Link className="flex" to={'/terms'}>
```
Comment (zavelevsky) [inline src/components/core/menu/mainMenu/MainMenuRightBurger/useBurgerMenuItems.tsx:61]: why not paths.privacy?

**Pull request merged**